### ISSUE TITLE
Support for MetricsCollection

### ISSUE THREAD

**Issue by eytan-avisror** [keikoproj/instance-manager#53]
Currently spec does not allow enabling MetricsCollection.
We should probably add support for it.

More info in the API docs:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-metricscollection.html

**Comment by RiyaJohn** [comment]
Can I try this issue?

**Comment by eytan-avisror** [comment]
Hi @RiyaJohn 
Sure! happy to take contributions.

I think #54 is very similar but is more urgently needed.

In any case feel free to work on either, I will add some more details to both issues

**Comment by eytan-avisror** [comment]
Implementation notes:

- We need to extend the InstanceGroup API to take `.spec.eks-cf.configuration.metrics` which is a list of metrics which should be added to the cloudformation template under a new object.

- In the current instance-manager config map that is in `config/crd/bases/instance-manager-configmap.yaml` there is a config map that holds the cloudformation template being used. we should modify the section that creates the autoscaling group and add templating for the MetricsCollection.

```
NodeGroup:
        Type: AWS::AutoScaling::AutoScalingGroup
        Properties:
          ...
          ...
          <we should add the MetricsCollection object here via go templating, for example>
          MetricsCollection:
            Metrics:
              - <user-requested-metric-1>
              - <user-requested-metric-1>
            Granularity: "1Minute" >> this is the only allowed value - we can hard code it
```

- Make sure to also take your changes to `docs/04_instance-manager.yaml` which has a copy of the configmap

- It will be nice to check whether provided metrics are one of allowed values and fail if it's not, however not sure current structure has a good place to add this in, we can do this with this PR or a future PR.

- Another nice-to-have is support for metric called `all` which would simply add all of the supported metrics

Essentially a user will provide input via the CR like so:
```
apiVersion: instancemgr.keikoproj.io/v1alpha1
kind: InstanceGroup
metadata:
  name: hello-world
spec:
  provisioner: eks-cf
  strategy:
    ...
  eks-cf:
    ...
    configuration:
      ...
      metrics:
        - groupMinSize
        - groupMaxSize
        - groupDesiredCapacity
        - groupInServiceInstances
        - groupPendingInstances
        - groupStandbyInstances
        - groupTerminatingInstances
        - groupTotalInstances
```

Which would change the cloudformation template to:

```
...
NodeGroup:
        Type: AWS::AutoScaling::AutoScalingGroup
        Properties:
          ...
          ...
          <we should add the MetricsCollection object here via go templating, for example>
          MetricsCollection:
            Metrics:
              - GroupMinSize
              - GroupMaxSize
              - GroupDesiredCapacity
              - GroupInServiceInstances
              - GroupPendingInstances
              - GroupStandbyInstances
              - GroupTerminatingInstances
              - GroupTotalInstances
            Granularity: "1Minute"
...
```

**Comment by RiyaJohn** [comment]
Thank you @eytan-avisror for the details. 
I started working on it today https://github.com/RiyaJohn/instance-manager/pull/1/files. Let me know if I am not going in the right direction.
Should I be following this https://github.com/keikoproj/instance-manager/blob/master/.github/DEVELOPER.md for setting it up on local for development?

**Comment by eytan-avisror** [comment]
@RiyaJohn yes, that doc should get you started with creating a vanilla EKS cluster and running instance-manager locally on it.

**Comment by RiyaJohn** [comment]
But the link to setup.sh file in the doc gives a 404 not found error

**Comment by eytan-avisror** [comment]
@RiyaJohn this could be related to your working directory, the file being referenced is this one:
https://github.com/keikoproj/instance-manager/blob/master/test-bdd/setup/setup.sh

**Comment by RiyaJohn** [comment]
Ok cool

**Comment by eytan-avisror** [comment]
closed by #57
